service/loader/checkpoint.go

```diff
@@ -44,7 +44,7 @@ func FromCheckpointFile(filepath string, log *zerolog.Logger) *Checkpoint {
 // Trie loads the execution state trie from the LedgerWAL root checkpoint.
 func (c *Checkpoint) Trie() (*trie.MTrie, error) {
 
-	trees, err := wal.LoadCheckpoint(c.filepath, c.log)
+	trees, err := wal.LoadCheckpoint(c.dir, c.log)
```

Comment:
we should keep the original change, it will still take the full path of the checkpoint file, but it will automatically split the dir and file name from it and find other files (i.e. root.checkpoint.0) in the same directory.